torch/_export/non_strict_utils.py

```diff
@@ -226,26 +232,53 @@ def make_constraints(
     if constraint_violation_error:
         raise constraint_violation_error
 
+    # create mapping from tensor id -> (dim -> user-specified constraint spec) for user tensor inputs
+    user_tensor_input_names = {
+        spec.arg.name
+        for spec in input_specs
+        if spec.kind == InputKind.USER_INPUT and isinstance(spec.arg, TensorArgument)
+    }
+    tensor_id_to_dim_constraint: Dict[int, Dict[int, Dict[str, int]]] = defaultdict(
+        defaultdict
+    )
+    for const in constraints:
+        spec = const.serializable_spec
+        tensor_id_to_dim_constraint[spec["t_id"]][spec["dim"]] = {
+            "lower": spec["min"],
+            "upper": spec["max"],
+        }
+
     range_constraints = {}
     input_dims = defaultdict(list)
     free_symbols = set()
+    input_index = 0
     for node in gm.graph.nodes:
-        if node.op != "placeholder":
+        if node.name not in user_tensor_input_names:
             continue
         if _is_constant_argument(node.meta["val"]) or isinstance(
             node.meta["val"], CustomObjArgument
         ):
             continue
+        tensor_constraints = tensor_id_to_dim_constraint[id(flat_args[input_index])]
         for i, d in enumerate(node.meta["val"].shape):
             if isinstance(d, torch.SymInt):
                 # Look up the range constraint for the symbol corresponding to this shape dimension
                 # and store it indexed by the symbolic expression corresponding to it.
                 # NOTE(avik): Use node._expr instead of node.expr for the lookup here because
                 # we want the symbol, not its replacement, which could be an expression. Maybe
                 # there's a better way to do this, e.g., by (re)computing value ranges for expressions?
-                range_constraints[d.node.expr] = shape_env.var_to_range[d.node._expr]
+                constraint = tensor_constraints[i] if tensor_constraints else None
```

Comment:
I'm not entirely sure what you're doing here. What makes it necessary to look up the specified range instead of the inferred range here? I think it is because the inferred range min is clamped to 2, have you investigated how to avoid that? I continue to be worried that this is not desirable.

@ezyang might need your help here. Is there something we can do (e.g., mark dynamic dims to be size-like) to avoid this clamping. Is the 0/1 specialization flag + default_range to blame here?